lib/api/v1/forms/upload_keywords.go

```diff
@@ -0,0 +1,80 @@
+package forms
+
+import (
+	"encoding/csv"
+	"errors"
+	"io"
+	"mime/multipart"
+
+	"github.com/markgravity/golang-ic/database"
+	"github.com/markgravity/golang-ic/lib/jobs"
+	"github.com/markgravity/golang-ic/lib/models"
+
+	"github.com/fatih/structs"
+)
+
+type UploadKeywordsForm struct {
+	FileHeader *multipart.FileHeader `form:"file" binding:"required"`
+	User       *models.User
+}
+
+func (f *UploadKeywordsForm) Save() error {
+	db := database.GetDB()
+	keywords, err := f.readCSVFile()
+	if err != nil {
+		return err
+	}
+
+	for _, k := range keywords {
+		keyword := &models.Keyword{
+			Text: k,
+			User: f.User,
+		}
+
+		err := keyword.Save(db)
+		if err != nil {
+			return err
+		}
+
+		job := jobs.Crawl{}
+		job.SetArgs(structs.Map(keyword))
+		err = jobs.Dispatch(&job)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Is there a way to save ALL the keywords in the DB first, in a single SQL query?
Then we can have a loop for triggering the jobs.

Creating, opening, running, parsing result and finaly closing an SQL transaction cost a lot of resources, so as much as possible, we must consider reducing the numbers of transactions 

More in the doc: https://gorm.io/docs/create.html#Batch-Insert :)